### PR TITLE
(LTH-1) Support 'make install' in leatherman

### PR DESCRIPTION
**Pull request by branan**
This adds the frameworks necessary to be able to install leatherman, as well as load it with the normal CMake `find_package` facilities. A companion PR to facter demonstrates the usage.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,3 +61,19 @@ if(LEATHERMAN_ENABLE_TESTING)
...
+    set(CMAKE_FILES
+        cmake/cflags.cmake
+        cmake/leatherman.cmake
+        cmake/options.cmake
+        cmake/leatherman_config.cmake
+    )
+    install(FILES ${CMAKE_FILES} DESTINATION "lib/cmake/leatherman/cmake/")
+
+    configure_file(LeathermanConfig.cmake.in "${PROJECT_BINARY_DIR}/LeathermanConfig.cmake" @ONLY)
+    message(STATUS ${PROJECT_BINARY_DIR})
```
Comment (peterhuene) [inline CMakeLists.txt]: Debugging message or intentional?  Guessing the former given it's just a path being output.

Comment (branan) [inline CMakeLists.txt]: Whoops, that was a debugging message. Will remove.

File: CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +21,9 @@ endif()
 defoption(LEATHERMAN_DEFAULT_ENABLE "Should Leatherman libraries all be built by default" ${LEATHERMAN_TOPLEVEL})
 defoption(LEATHERMAN_DEBUG "Enable verbose logging messages from leatherman macros" FALSE)
 defoption(LEATHERMAN_ENABLE_TESTING "Build the leatherman test binary" ${LEATHERMAN_DEFAULT_ENABLE})
+defoption(LEATHERMAN_INSTALL "Install the leatherman libraries and headers" ${LEATHERMAN_DEFAULT_ENABLE})
+
+set(BUILDING_LEATHERMAN TRUE)
```
Comment (MikaelSmith) [inline CMakeLists.txt]: When is this ever false?

Comment (MikaelSmith) [inline CMakeLists.txt]: Oh I see, it's so all the library cmake files can be re-used for configuration.

**Comment by MikaelSmith** [comment]
Looks good; I'm going to play with installing it a bit.

**Comment by MikaelSmith** [comment]
:+1:

File: CMakeLists.txt (hunk 2 above)
Comment (branan) [inline CMakeLists.txt]: Yup. I'm going to update the commit message and the README with more info on how/why this is the case.

**Comment by branan** [comment]
README updated. I don't think there are any other changes I want to make to this, so once the README is +1'd we're good to go.

The only change in the updated base commit was the removal of the debug message Peter found.

File: README.md [hunk 3]
```diff
@@ -52,6 +87,24 @@ variables:
 * `LEATHERMAN_<LIBRARY>_LIBS` The contents of both
   `LEATHERMAN_<LIBRARY>_LIB` and `LEATHERMAN_<LIBRARY>_DEPS`
 
+### CMake Helpers Provided by Leatherman
+
+In addition to the C++ library components, Leatherman provides a few
+CMake helpers. These will be automatically added to your
+`CMAKE_MODULE_PATH` when `find_package` is processed.
```
Comment (branan) [inline README.md]: I'm kind of tempted to just include `options`, `cflags`, and `leatherman` in the Finder and obviate the need to explicitly include them. I can't think of a situation where we *wouldn't* want them to be there.

My only concern with that is certain things aren't well-namespaced. I may hold off on that until a separate PR where I can rename all the macros to have a `leatherman_` or `lth_` prefix.

**Comment by branan** [comment]
@MikaelSmith Mentioned Boost in a new dependencies section, and added a note in the "Using Leatherman" section about `CMAKE_PREFIX_PATH`

**Comment by MikaelSmith** [comment]
Typo in the commit message.

**Comment by branan** [comment]
@MikaelSmith annnnd fixed

**Pull request merged**